src/compiler/crystal/semantic/call.cr

```diff
@@ -122,7 +122,7 @@ class Crystal::Call
   end
 
   def lookup_matches(*, with_literals = false)
-    if args.any? { |arg| arg.is_a?(Splat) || arg.is_a?(DoubleSplat) }
+    if args.any?(Splat | DoubleSplat)
```

Comment:
This will collapse into the superclass and match everything.

Reply:
Wow